.circleci/workflow-queue-v1.sh

```diff
@@ -100,30 +111,48 @@ if [[ ! "${current_pipeline_start_time}" ]]; then
   exit 1
 fi
 
+fetch_older_pipelines "${current_pipeline_start_time}"
+pipeline_workflow_ids=$__
+printf "%s\n%s\n\n" "Currently running workflow_id are:" "${pipeline_workflow_ids}"
+
+if [[ -z $pipeline_workflow_ids ]]; then
+  printf "%s\n" "No workflow currently running."
+  exit 0
+fi
+
 # Wait as long as "pipelines" variable is not empty until max time has reached.
+# Max wait time until workflows have finished is 45 minutes because e2e tests may take a long time to finish.
+# DISCLAIMER This max time may not be enough.
+max_time=$((45 * 60))
 is_running=true
 waited_time=0
-wait="30s"
+sleep_time="10s"
+sleep_time_counter=10
 
 while [[ "${is_running}" == "true" ]]; do
   printf "\n***\n"
-  fetch_older_pipelines "${current_pipeline_start_time}"
-  pipelines=$__
-
-  if [[ $pipelines ]]; then
-    printf "\n%s\n" "${pipelines}"
-    printf "%s\n" "Waiting for previously submitted pipelines to finish. sleep ${wait} seconds. Please wait..."
-    sleep $sleep_time
-    waited_time=$((sleep_time + waited_time))
-  else
-    printf "\n%s\n" "All previously submitted pipelines have finished."
+
+  for id in ${pipeline_workflow_ids}; do
     is_running=false
-  fi
+    fetch_pipeline_jobs "${id}"
```

Comment:
I don't understand what the bug was and how this fixes it, can you add a comment here that explains what is happening?